kroxylicious/src/main/java/io/kroxylicious/proxy/internal/codec/KafkaRequestDecoder.java

```diff
@@ -85,12 +89,19 @@ protected Frame decodeHeaderAndBody(ChannelHandlerContext ctx, ByteBuf in, final
 
         RequestHeaderData header = null;
         final ByteBufAccessorImpl accessor;
+        Metrics.globalRegistry.counter("inbound_downstream_messages").increment();
         var decodeRequest = decodePredicate.shouldDecodeRequest(apiKey, apiVersion);
         LOGGER.debug("Decode {}/v{} request? {}, Predicate {} ", apiKey, apiVersion, decodeRequest, decodePredicate);
         boolean decodeResponse = decodePredicate.shouldDecodeResponse(apiKey, apiVersion);
         LOGGER.debug("Decode {}/v{} response? {}, Predicate {}", apiKey, apiVersion, decodeResponse, decodePredicate);
         short headerVersion = apiKey.requestHeaderVersion(apiVersion);
         if (decodeRequest) {
+            Metrics.globalRegistry.counter("decoded_messages").increment();
+            Metrics.globalRegistry.counter("bytes_read").increment(length);
+            Metrics.globalRegistry.summary("message_size", List.of(
```

Comment:
I think `request_size` would be a better name.

Reply:
Also, shouldn't we record a summary for the response_size too (i.e. `KafkaResponseDecoder`)

Reply:
Noted - I'm hoping where I've added the metric into `KafkaResponseDecoder` is the right spot - @SamBarker? :)

Reply:
@racheljpg you haven't pushed a change to `KafkaResponseDecoder` so I can't tell if its in the right spot. 

(Sorry I've had this tab open for days without actually sending the comment)

Reply:
thanks Sam, it is there now :)